test/unit/ngsiv2/amqpBinding-test.js

```diff
@@ -0,0 +1,320 @@
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[iot_support@tid.es]
+ *
+ * Modified by: Fernando Méndez, Daniel Calvo - ATOS Research & Innovation
+ */
+
+'use strict';
+
+var iotagentUl = require('../../../'),
+    config = require('./config-test.js'),
+    nock = require('nock'),
+    async = require('async'),
+    request = require('request'),
+    utils = require('../../utils'),
+    iotAgentLib = require('iotagent-node-lib'),
+    amqp = require('amqplib/callback_api'),
+    apply = async.apply,
+    contextBrokerMock,
+    amqpConn,
+    channel;
+
+function startConnection(exchange, callback) {
+    amqp.connect('amqp://localhost', function(err, conn) {
+        amqpConn = conn;
+
+        conn.createChannel(function(err, ch) {
+            ch.assertExchange(exchange, 'topic', {});
+
+            channel = ch;
+            callback(err);
+        });
+    });
+}
+
+describe('AMQP Transport binding: measures', function() {
+    beforeEach(function(done) {
+        var provisionOptions = {
+            url: 'http://localhost:' + config.iota.server.port + '/iot/devices',
+            method: 'POST',
+            json: utils.readExampleFile('./test/unit/ngsiv2/deviceProvisioning/provisionDeviceAMQP1.json'),
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        nock.cleanAll();
+
+        // Note: /v1/updateContext response is not processed by IOTA so its content is irrelevant,
+        // as far as it is a 200 OK
+        contextBrokerMock = nock('http://192.168.1.1:1026')
+            .matchHeader('fiware-service', 'smartGondor')
+            .matchHeader('fiware-servicepath', '/gardens')
+            .post('/v1/updateContext')
+            .reply(200, '{}');
+
+        async.series([
+            apply(iotagentUl.start, config),
+            apply(request, provisionOptions),
+            apply(startConnection, config.amqp.exchange)
+        ], done);
+    });
+
+    afterEach(function(done) {
+        nock.cleanAll();
+
+        amqpConn.close();
+
+        async.series([
+            iotAgentLib.clearAll,
+            iotagentUl.stop
+        ], done);
+    });
+
+    describe('When a new single measure arrives to a Device routing key', function() {
+        beforeEach(function() {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20UL%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/singleMeasure.json'))
+                .reply(204);
+        });
+
+        it('should send a new update context request to the Context Broker with just that attribute', function(done) {
+            channel.publish(config.amqp.exchange, '.1234.MQTT_2.attrs.temperature', new Buffer('23'));
```

Comment:
Not sure if I'm undertanding correctly the script of this tests...

As far as I understand, you are seding temperature = 23 using AMQP binding. However, the singleMeasure.json which you program in the contextBrokerMock() is for hummidy = "98".

So, I'm missing something? :)

Reply:
I have realized what I missing... I was looking to *second*SingleMeasure.json, not *singleMeasure.json".

NTC... sorry the noise ;)